src/main/java/com/work/management/service/impl/ProjectService.java

```diff
@@ -0,0 +1,54 @@
+package com.work.management.service.impl;
+
+import com.work.management.entity.Project;
+import com.work.management.repository.EmployeeRepository;
+import com.work.management.repository.ProjectRepository;
+import com.work.management.web.rest.resource.ProjectResource;
+import com.work.management.service.DataPersistingService;
+import com.work.management.utils.ExceptionUtils;
+import lombok.NoArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+import java.util.Objects;
+
+@Service
+public final class ProjectService implements DataPersistingService<ProjectResource, Project> {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+
+    private final ProjectRepository projectRepository;
+    private final EmployeeRepository employeeRepository;
+
+    public ProjectService(ProjectRepository projectRepository, EmployeeRepository employeeRepository) {
+        this.projectRepository = projectRepository;
```

Comment:
Remove public from constructor and from class name also...make it package private.
Read: https://github.com/google/guice/wiki/KeepConstructorsHidden

Reply:
Because you are using dependency injection you don't need to make your class publically accessible by making the constructor and class public. By making it package private you make sure that everyones who want's a instance of your class will need to use dependency injection in order to do so. Just the interface needs to be public, and nobody should know which class is implementing this interface or should not get hold of the implementation class, that's why were are implementing a interface.

Reply:
Nice point i was not aware about this